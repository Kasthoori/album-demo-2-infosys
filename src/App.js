import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Header from './components/header/Header';
import { Table } from 'react-bootstrap';
import ListItem from './components/ListItem/ListItem';
import { useHistory } from 'react-router-dom';




const App = () => {
  
  const [albums, setAlbums] = useState([]);
  const [tracks, setTracks] = useState([]);
  const history = useHistory();

  const url1 = "http://jsonplaceholder.typicode.com/albums";
  const url2 = "http://jsonplaceholder.typicode.com/photos";

  document.title = 'Albums Available';

  useEffect(() => {
    async function fetchData() {
      const resp_albums = await axios.get(url1);
      const resp_tracks = await axios.get(url2);

      setAlbums(resp_albums.data);
      setTracks(resp_tracks.data);
    }

    fetchData();
  }, []);

  const listOnClick = (id, title, userId) => {
    let tracks_in_album = tracks.filter((item) => {
      return item.albumId === id;
    });

    history.push("/details", {
      status: "true",
      id: id,
      title: title,
      userId: userId,
      tracks: tracks_in_album,
    });
  };


console.log("test data check", albums);

  return (
    <div className="App">
      <span data-testid="headercomp"><Header /></span>
        <br />
        <br />
      <div className="container">
        <span data-testid="datatables">
          <Table table table-striped>
            <thead>
              <tr>
                <th data-testid="albumid">Album ID</th>
                <th data-testid="albumtitle">Album Title</th>
                <th data-testid="moredetails">More Details</th>
              </tr>
            </thead>
            <tbody>
            
            {albums.map((item) => (
              <ListItem
                key={item.id + item.userId}
                id={item.id}
                title={item.title}
                userId={item.userId}
                onClick={listOnClick}
              />
            ))}

            </tbody>
          </Table>
        </span>
      </div>
    </div>
  );
}

export default App;
