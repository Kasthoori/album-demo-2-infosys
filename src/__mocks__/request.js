const albums = {

    4: {title: 'Album One'},
    5: {tilte: 'Album Two'},
  };
  
  export default function request(url) {
    return new Promise((resolve, reject) => {
      const albumID = parseInt(url.substr('/albums/'.length), 10);
      process.nextTick(() =>
        albums[albumID]
          ? resolve(albums[albumID])
          : reject({
              error: 'User with ' + albumID + ' not found.',
            }),
      );
    });
  }