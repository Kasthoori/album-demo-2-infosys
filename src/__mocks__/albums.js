import request from './request';

export async function getAlbumTitle(albumID) {
  return request('/albums/' + albumID).then(album => album.title);
}