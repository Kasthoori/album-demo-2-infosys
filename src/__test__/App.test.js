/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
import { cleanup, wait, render } from '@testing-library/react';
import * as albums from '../__mocks__/albums';
import axios from 'axios';
import toJson from "enzyme-to-json";




describe('App Component Test', () => {

  jest.mock('../__mocks__/albums');
  jest.mock('axios');

  afterEach(cleanup);
 

  //test case 01. render header component

      test('render header component', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.find('[data-testid="headercomp"]').length).toBe(1);
      });

  //test case 02. url call of fetchData

      it('mocking data fetching url call',async () => {
        expect.assertions(1);
        return albums.getAlbumTitle(4).then(data => expect(data).toEqual('Album One'));
      });

  //test case 03. axios cass in fetchData function
      test('mocking axios request', async () => {
        const data = {
        data: [
              {
                id: 1,
                title: 'title 1'
              },
              {
                id: 2,
                title: 'title 2'
              },
              {
                id: 3,
                title: 'mocked title'
              }
            ]
        };
        
        axios.get.mockResolvedValueOnce(data);
        const wrapper = shallow(<App />);
            await wait(() => {
              expect(wrapper.exists()).toEqual(true);
            });
        });

  //test case 04. render data table

      test('should render data table', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.find('[data-testid="datatables"]').length).toBe(1);
      });

  //test case 05 number of rendering columns
      test('number of rendering columns', () => {
        const wrapper = shallow(<App  />);
        expect(wrapper.find('th').length).toBe(3);
        expect(wrapper.find('[data-testid="albumid"]').text()).toBe('Album ID');
        expect(wrapper.find('[data-testid="albumtitle"]').text()).toBe('Album Title');
        expect(wrapper.find('[data-testid="moredetails"]').text()).toBe('More Details');

      });

      //test case 06 to chect withou crashing (snapshot)
      test('render without crashing', () => {
          const wrapper = shallow(<App  /> );
          expect(toJson(wrapper)).toMatchSnapshot();
        
      
      });

   

});

