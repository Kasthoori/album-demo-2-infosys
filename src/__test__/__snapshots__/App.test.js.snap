// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Test render without crashing 1`] = `
<div
  className="App"
>
  <span
    data-testid="headercomp"
  >
    <Header />
  </span>
  <br />
  <br />
  <div
    className="container"
  >
    <span
      data-testid="datatables"
    >
      <ForwardRef
        table={true}
        table-striped={true}
      >
        <thead>
          <tr>
            <th
              data-testid="albumid"
            >
              Album ID
            </th>
            <th
              data-testid="albumtitle"
            >
              Album Title
            </th>
            <th
              data-testid="moredetails"
            >
              More Details
            </th>
          </tr>
        </thead>
        <tbody />
      </ForwardRef>
    </span>
  </div>
</div>
`;
